src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java

```diff
@@ -180,6 +199,77 @@ public void updateTokenStats() {
             }
         });
     }
+    
+    /**
+     * {@code GET /user-usage-analysis}: Analyze user usage
+     * 
+     * @throws IOException
+     */
+    @GetMapping(path = "/user-usage-analysis")
+    public void analyzeUserUsage() throws IOException {
+        log.info("User usage analysis started...");
+        List<UserTokenUsageWithInfo> tokenStats =  tokenStatsService.getTokenUsageAnalysis(Instant.now().minus(365, ChronoUnit.DAYS));    
+        UsageSummary resourceSummary = new UsageSummary();       
+        Map<Long, UserUsage> userSummary = new HashMap<>();
+
+        for (UserTokenUsageWithInfo state: tokenStats){
+            calculateUsageSummary(resourceSummary, state);
+
+            long userId = state.getToken().getUser().getId();
+            User user = state.getToken().getUser();
+            if (!userSummary.containsKey(userId)){
+                userSummary.put(userId, new UserUsage());
+                UserUsage currentUsage = userSummary.get(userId);
+                currentUsage.setUserFirstName(user.getFirstName());
+                currentUsage.setUserLastName(user.getLastName());
+                currentUsage.setUserEmail(user.getEmail());
+                currentUsage.setSummary(new UsageSummary());
+                Optional<UserDetails> userDetails = userDetailsRepository.findOneByUser(user);
+                if (userDetails.isPresent()){
+                    currentUsage.setJobTitle(userDetails.get().getJobTitle());
+                    currentUsage.setCompany(userDetails.get().getCompany());
+                    currentUsage.setLicenseType(userDetails.get().getLicenseType().getName());
+                }
+            }
+            UsageSummary currentUsageSummary = userSummary.get(userId).getSummary();
+            calculateUsageSummary(currentUsageSummary, state);
+        }
+        
+        ObjectMapper mapper = new ObjectMapper();
+        File resourceResult = new File("./resourceSummary.json");
+        File userResult = new File("./userSummary.json");
+        mapper.writeValue(resourceResult, resourceSummary);
+        mapper.writeValue(userResult, userSummary);
+
+        String s3AccessKey = applicationProperties.getAws().getS3AccessKey();
+        String s3SecretKey = applicationProperties.getAws().getS3SecretKey();
+        String s3Region = applicationProperties.getAws().getS3Region();
+        AWSCredentials credentials = new BasicAWSCredentials(s3AccessKey, s3SecretKey);
+        AmazonS3 s3client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials)).withRegion(s3Region).build();
```

Comment:
We have quite few places using s3client now. Could you write a service for this? So we could wire it using beans?